File: AN_2/SEM_1/MAP/lab2/src/tests/StrategyTaskRunnerTest.java
package tests;

import model.MessageTask;
import decorator.StrategyTaskRunner;
import decorator.TaskRunner;
import Enum.Strategy;

public class StrategyTaskRunnerTest
{
    public static void strategyTaskRunnerTest(Strategy strategy, MessageTask[] messageTasks)
    {
        TaskRunner taskRunner = new StrategyTaskRunner(strategy);
        taskRunner.addTask(messageTasks[0]);
        taskRunner.addTask(messageTasks[1]);
        taskRunner.addTask(messageTasks[2]);
        taskRunner.executeAll();
    }
}

File: AN_2/SEM_1/MAP/lab9/lab9/Tests/StrategyTaskRunnerTest.cs
using Model;
using Decorator;
using Enum;

namespace Tests
{
    public class StrategyTaskRunnerTest
    {
        public static void RunStrategyTaskRunnerTest(Strategy strategy, MessageTask[] messageTasks)
        {
            var taskRunner = new StrategyTaskRunner(strategy);
            taskRunner.AddTask(messageTasks[0]);
            taskRunner.AddTask(messageTasks[1]);
            taskRunner.AddTask(messageTasks[2]);
            taskRunner.ExecuteAll();
        }
    }
}
